### PR TITLE
Is this the right track?

### PR DESCRIPTION
**Pull request by HDCharles**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #55852 testing diff
* **#55671 Is this the right track?**

Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

### CODE REVIEW

**Comment by vkuzo** [comment]
this is a good start!  I have two high level comments:
1. since we don't have a fast kernel, I think it would be good to have this off by default, and we can let the user enable it manually.  We still want the fp32 gelu to be the default for all models, unless they opt in to this reference kernel.  We can probably just not register the handler, and add a comment on why it's not registered, unless @jerryzh168  has a cleaner way.
2. we should add at least two tests: (a) testing that with default settings, we still get the old gelu, and (b) testing that if enabled, this kernel is swapped, and the numerics are correct.

File: torch/quantization/fx/quantization_patterns.py [hunk 1]
```diff
@@ -1056,3 +1056,34 @@ def convert(self, quantizer: QuantizerCls, node: Node, load_arg: Callable,
...
+            self.gelu = quantizer.modules[self.gelu_node.target]  # next node?
+        elif node.op == "call_function":
+            self.gelu = node.target
+
+    def convert(self, quantizer: QuantizerCls, node: Node, load_arg: Callable,
+                is_reference: bool = False,
+                convert_custom_config_dict: Dict[str, Any] = None) -> Node:
+        if not is_reference:
+            warnings.warn("Only reference implementation support at this time")
+        args = load_arg(quantized=[0, 1])(self.gelu_node.args)
```
Comment (jerryzh168) [inline torch/quantization/fx/quantization_patterns.py:1076]: gelu only has one argument I think? https://pytorch.org/docs/stable/nn.functional.html#torch.nn.functional.gelu

you can do load_arg(quantized=[0])

File: torch/quantization/fx/quantization_patterns.py [hunk 2]
```diff
@@ -1056,3 +1056,34 @@ def convert(self, quantizer: QuantizerCls, node: Node, load_arg: Callable,
...
+        args = load_arg(quantized=False)(self.gelu_node.args)
+        kwargs = load_arg(quantized=False)(self.gelu_node.kwargs)
+        op_out = quantizer.quantized_graph.create_node(
+            "call_function", self.gelu, args, kwargs)
```
Comment (jerryzh168) [inline torch/quantization/fx/quantization_patterns.py:1080]: you can use functional for both call module and call function but target be torch.nn.functional.gelu I think
```
op_out = quantizer.quantized_graph.create_node(
            "call_function",  torch.nn.functional.gelu, args, kwargs)
```

**Comment by jerryzh168** [comment]
> this is a good start! I have two high level comments:
> 
> 1. since we don't have a fast kernel, I think it would be good to have this off by default, and we can let the user enable it manually.  We still want the fp32 gelu to be the default for all models, unless they opt in to this reference kernel.  We can probably just not register the handler, and add a comment on why it's not registered, unless @jerryzh168  has a cleaner way.
> 2. we should add at least two tests: (a) testing that with default settings, we still get the old gelu, and (b) testing that if enabled, this kernel is swapped, and the numerics are correct.

not enabling this by default sounds good to me if performance is a concern. although I think we are planning to change quantization to produce reference pattern by default, and will have another transformation to transform the reference model to the final backend specific model (fbgemm/qnnpack etc.), at that time I think we should produce reference pattern by default.

**Pull request closed without merging**